scripts/sanity_runnable.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -eu
+
+TEST_PROJECT="smoketest"
+
+trap 'catch $? $LINENO' EXIT
+
+catch() {
+  if [[ "$1" != "0" ]]; then
+    echo "Error $1 occurred on $2"
+  fi
+
+  # return to origin, clear directory stack
+  pushd -0 > /dev/null && dirs -c
+
+  if [[ -d "$TEST_PROJECT" ]]; then
+    echo "cleaning up test artifacts"
+    rm -rf "$TEST_PROJECT"
+  fi
+}
+
+# create a new project
+subo create project "$TEST_PROJECT"
+
+# enter project directory
+pushd "$TEST_PROJECT" > /dev/null
+
+# create a runnable for each supported language
+subo create runnable rs-test --lang rust
+subo create runnable swift-test --lang swift
+subo create runnable as-test --lang assemblyscript
+subo create runnable tinygo-test --lang tinygo
+subo create runnable js-test --lang javascript
```

Comment:
if my understanding is correct, if this completes successfully, the files would be left in place, rather than cleaned up after.

Is that expected / desired?

Reply:
The trap is triggered on exit (error or otherwise) which cleans up the root project directory if present.

set -e tells the script to terminate on error, set -u tells the script to error if a variable isn't bound to a value (useful when you have potentially dangerous commands like `rm -rf $var`

Reply:
I do need to comment out javascript for  now since it will fail to bundle until the new images are released

Reply:
Had a conversation on discord, the `trap` takes care of running the `catch` function to clean up whatever happens (error or success)